app/agent/controllers/business/BusinessAccountingPeriodDateController.scala

```diff
@@ -74,31 +74,34 @@ class BusinessAccountingPeriodDateController @Inject()(val baseConfig: BaseContr
           isEditMode = isEditMode
         )),
         accountingPeriod =>
-          if (accountingPeriodService.checkEligibleAccountingPeriod(accountingPeriod.startDate.toLocalDate, accountingPeriod.endDate.toLocalDate)) {
-            for {
-              cache <- keystoreService.fetchAll() map (_.get)
-              optOldAccountingPeriodDates = cache.getAccountingPeriodDate()
-              _ = cache.getIncomeSource() map (source => IncomeSourceType(source.source))
-              _ <- keystoreService.saveAccountingPeriodDate(accountingPeriod)
-              taxYearChanged = optOldAccountingPeriodDates match {
-                case Some(oldAccountingPeriod) => (oldAccountingPeriod.taxEndYear != accountingPeriod.taxEndYear)
-                case None => true
-              }
-              _ <- if (taxYearChanged) keystoreService.saveTerms(terms = false)
-              else Future.successful(Unit)
-            } yield {
-              if (isEditMode) {
-                if (taxYearChanged) {
-                  Redirect(agent.controllers.routes.TermsController.show(editMode = true))
+          keystoreService.fetchIncomeSource() flatMap { incomeSources =>
+            if (accountingPeriodService.checkEligibleAccountingPeriod(accountingPeriod.startDate.toLocalDate,
+              accountingPeriod.endDate.toLocalDate, incomeSources.contains(IncomeSourceType(IncomeSourceType.both)))) {
```

Comment:
```suggestion
              accountingPeriod.endDate.toLocalDate, incomeSources.contains(Both))) {
```
Would it be better to compare against the actual case object?